src/Microsoft.DotNet.Build.Tasks/GenerateBindingRedirect.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.IO;
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+using Task = Microsoft.Build.Utilities.Task;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class GenerateBindingRedirect : Task
+    {
+        [Required]
+        public ITaskItem[] AssemblyIdentities { get; set; }
+
+        [Required]
+        public ITaskItem[] Executables { get; set; }
+
+        [Required]
+        public string OutputPath { get; set; }
+
+        private static XNamespace ns { get; set; }
+        public override bool Execute()
+        {
+            ns = "urn:schemas-microsoft-com:asm.v1";
+            XElement bindingRedirectAssemblies = new XElement(ns + "assemblyBinding");
+            foreach (ITaskItem assembly in AssemblyIdentities)
+            {
+                string assemblyVersion = assembly.GetMetadata("Version");
+                XElement assemblyIdentity = new XElement("assemblyIdentity",
+                                                         new XAttribute("name", assembly.GetMetadata("Name")),
+                                                         new XAttribute("publicKeyToken", assembly.GetMetadata("PublicKeyToken")),
+                                                         new XAttribute("culture", string.IsNullOrEmpty(assembly.GetMetadata("Culture")) ? "neutral" : assembly.GetMetadata("Culture")));
+                assemblyIdentity = RemoveNamespace(assemblyIdentity);
+                XElement bindingRedirect = new XElement("bindingRedirect",
+                                                        new XAttribute("oldVersion", $"0.0.0.0-{assemblyVersion}" ),
+                                                        new XAttribute("newVersion", assemblyVersion));
+                bindingRedirect = RemoveNamespace(bindingRedirect);
+                XElement dependentAssembly = new XElement("dependentAssembly",
+                                                          assemblyIdentity,
+                                                          bindingRedirect);
+                dependentAssembly = RemoveNamespace(dependentAssembly);
+                bindingRedirectAssemblies.Add(dependentAssembly);
+            }
+            XDocument doc = new XDocument(new XElement("configuration", new XElement("runtime", bindingRedirectAssemblies)));
+            foreach (ITaskItem executable in Executables)
+            {
+                string executableName = Path.GetFileName(executable.ItemSpec);
+                using (FileStream fs = new FileStream(Path.Combine(OutputPath, executableName+".config"), FileMode.Create))
+                {
+                    doc.Save(fs);
+                }
+            }
+            return true;
```

Comment:
as general good practice Execute() on all MSBuild tasks should return thus
`return !Log.HasLoggedErrors;`

In this case it will be true anyway -- but are there any exceptions/error cases to handle and log here?